// karma/bin/karma.conf.js
// Karma configuration
// Generated on Thu Feb 09 2017 17:13:20 GMT+0530 (India Standard Time)

module.exports = function(config) {
  config.set({

    // base path that will be used to resolve all patterns (eg. files, exclude)
    basePath: '',


    // frameworks to use
    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
     frameworks: ['jasmine'],


    // list of files / patterns to load in the browser
    files: [
		'/wamp/www/Jasmine-Examples-New/lib/jquery-3.1.1.min.js', 
		'/wamp/www/Jasmine-Examples-New/lib/jasmine-jquery.js', 
		'/wamp/www/Jasmine-Examples-New/src/required-config.js', 
		'/wamp/www/Jasmine-Examples-New/src/custom.js', 
		'/wamp/www/Jasmine-Examples-New/spec/customSpec.js', 
		{pattern: '/wamp/www/Jasmine-Examples-New/spec/*Spec.js', included: false}  
    ],
	

    // list of files to exclude
    exclude: [
    ],


    // preprocess matching files before serving them to the browser
    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
    preprocessors: {
    },


    // test results reporter to use
    // possible values: 'dots', 'progress'
    // available reporters: https://npmjs.org/browse/keyword/karma-reporter


    reporters: ["spec"],
    specReporter: {
        maxLogLines: 5,         // limit number of lines logged per test
        suppressErrorSummary: true,  // do not print error summary
        suppressFailed: false,  // do not print information about failed tests
        suppressPassed: false,  // do not print information about passed tests
        suppressSkipped: true,  // do not print information about skipped tests
        showSpecTiming: false // print the time elapsed for each spec
      }, 


    // web server port
    port: 9876,


    // enable / disable colors in the output (reporters and logs)
    colors: true,


    // level of logging
    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
    logLevel: config.LOG_ERROR,
    browserConsoleLogOptions: {
      terminal: false
    },


    // enable / disable watching file and executing tests whenever any file changes
    autoWatch: true,

     // start these browsers
    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
	browsers: ['Chrome'],
    browsers: ['Chrome_without_security'],
	customLaunchers: {
      Chrome_without_security: {
        base: 'Chrome',
        flags: ['--disable-web-security']
      }
    }, 
	
	plugins: [
        'karma-jasmine-html-reporter',
        'karma-spec-reporter',
        'karma-chrome-launcher',
        'karma-jasmine',
        'karma-coverage',
        'karma-phantomjs-launcher'
    ],

    // Continuous Integration mode
    // if true, Karma captures browsers, runs the tests and exits
    singleRun: false,

    // Concurrency level
    // how many browser should be started simultaneous
    concurrency: Infinity
  })
}
